upup/pkg/fi/cloudup/bootstrapchannelbuilder.go

```diff
@@ -28,7 +28,8 @@ import (
 )
 
 type BootstrapChannelBuilder struct {
-	cluster *kops.Cluster
+	cluster   *kops.Cluster
+	Lifecycle *fi.Lifecycle
```

Comment:
I think bootstrap is pretty tightly bound to the rest of the cluster.  And if only the CNI has changed, then I think that our logic should ensure that only that is actually changed...